### ISSUE TITLE
Backdrop treats config field definitions like real fields

### ISSUE THREAD

**Issue by docwilmot** [backdrop/backdrop-issues#5170]
**Description of the bug**

In Drupal, a field would be created by [field_create_field()](https://api.drupal.org/api/drupal/modules%21field%21field.crud.inc/function/field_create_field/7.x), which function created both the field config (in database) and the field data tables at the same time. We've split in Backdrop, so that field config is now in files. 

But in Backdrop, if you simply copy a field config to the file folder, all the Field API functions work happily on it: `field_read_fields()`, `field_info_fields()` etc return appropriate responses, because all these functions only check if the config exists. You can even copy paste a field instance and your "field" will show up on an entity form. 

But if you try and save that entity form, an exception happens, complaining that the field data table does not exist.

**Steps To Reproduce**

To reproduce the behavior:
- copy a field config file (they start with `field.field.whatever.json`) and rename it, including the config name, in the json data. 
- nothing happens. Backdrop is quite happy with this fake field.
- call `field_read_fields()` using your fake field name. Success!
- copy and paste the field instance config as well (`field.instance.node.post.field_whatever.json`) and rename as before
- visit a Post node form; your field will be attached to the form! (clear caches first)
- try and save post.

**Actual behavior**

`SQLSTATE[42S02]: Base table or view not found: 1146 Table 'bd.field_data_field_whatever' doesn't exist`

**Expected behavior**

Either we should allow Backdrop to recognize when the field tables do not exist and return an error on Field API functions
Or
Let this be as is but add a key in the field info returned by those functions that tells devs all is not well (`tables_created: FALSE` for eg)
Or
If a user tries to save this field, then create the tables automatically on the fly. Not sure if this has implications I havent thought of, but I think this is my preference.

**Additional information**
v 1.19.0

**Comment by indigoxela** [comment]
Yes, that's a concept problem, I guess...

Warning devs on, for instance, the status page might be the quickest solution. But thinking this over - we shouldn't let users act on broken fields and run into an Exception because of that. So checking for table existence seems a cleaner solution to me.

What should it look like on UI?

**Comment by laryn** [comment]
> If a user tries to save this field, then create the tables automatically on the fly. Not sure if this has implications I havent thought of, but I think this is my preference.

@docwilmot  I'm also not sure if this has implications and also prefer it if it's possible. 

In my spare time  I'm trying to get a module working that allows selective installation of pre-configured Paragraph types (so not automatically activating them all from a `/config` folder but allowing a selection only of the ones that you need on that particular site). If I provide all the relevant configs (for fields, instances, Paragraph type, etc.) and save those configs into the site's configuration, could the relevant database table structure be built automatically from the config files?

**Comment by docwilmot** [comment]
If we get https://github.com/backdrop/backdrop-issues/issues/3224 done, then when you enable your module, ALL field configs in your module's `/config` folder will be fully synced and tables created, no choice. If that is not what you want (sounds like it isnt) then you'll need to store those configs in a directory thats NOT called `/config`. Then your module will need to call its own modified version of `config_install_default_config()` and invoke `hook_config_create()` in that.

**Comment by laryn** [comment]
Thanks, I'm picking through that one now. I have the configs in a separate folder, correct. Thanks!

**Comment by jenlampton** [comment]
Under steps to reproduce, this is the first item:

> copy a field config file (they start with field.field.whatever.json) and rename it, including the config name, in the json data.

This use case should never happen. 

The three supported ways to get config into the active directory are by doing a sync from staging (where the database structure will be created), installing a new module (where the database structure is created), or by adding the field from the field UI (where the database structure will be created). 

I don't think we should worry about people trying things that aren't supported... that could be a very slippery slope. I think we should close this issue unless there's a supported scenario where this could happen.

**Comment by docwilmot** [comment]
I agree-ish. The reason I'd keep this issue is that the `active` folder looks like where fields come from, to a person who doesnt know the "correct" options you list above, so it might seem reasonable to create a new field by just copying an old one. Agreed thats bad, but the the problem is that Backdrop encourages this by making it seem all OK until it crashes. So I'd think that we should some way warn people that theyve done a non-supported thing.

But I agree this is unlikely and probably not a priority.

**Comment by BWPanda** [comment]
Perhaps we can add a note to the README that's created in these directories...?

https://github.com/backdrop/backdrop/blob/1.x/core/includes/install.inc#L867-L870
